hs_core/models.py

```diff
@@ -1962,6 +1962,34 @@ def get_irods_storage(self):
         else:
             return IrodsStorage()
 
+    def set_quota_holder(self, setter, new_holder):
```

Comment:
@pkdash I initially implemented this method without setter, however, during the process when I updated access control tests, I found we need a setter to be passed in to set_quota_holder() so that it can be validated that the setter is actually an owner of the resource. Without this validation, there is a possibility a non-owner can call this method to set quota owner to another owner which is not allowed. So I think it is good to add this safe guard to make sure the user (i.e., setter) who requests to transfer quota holder to another user is actually an owner. Let me know if this does not make sense to you.